### ISSUE TITLE
[bug] zarf 0.37.0-r0 build version not set correctly

### ISSUE THREAD

**Issue by ericsmalling** [wolfi-dev/os#25140]
The zarf version info in the recently updated 0.37.0-r0 package is reporting `unset-development-only` instead of `0.37.0`.

```
eric-smalling:~/git/zarf-playground/k8ssandra$ sudo apk add zarf~0.36.1
(1/1) Installing zarf (0.36.1-r0)
OK: 2846 MiB in 188 packages
eric-smalling:~/git/zarf-playground/k8ssandra$ zarf version
v0.36.1

eric-smalling:~/git/zarf-playground/k8ssandra$ sudo apk add zarf~0.37.0
(1/1) Upgrading zarf (0.36.1-r0 -> 0.37.0-r0)
OK: 2846 MiB in 188 packages
eric-smalling:~/git/zarf-playground/k8ssandra$ zarf version

 NOTE  Using config file
unset-development-only
```
This also breaks functionality such as `zarf init` as it depends on that version info to pull the correct init package down.

```
zarf init
...
     ERROR:  failed to download the init package: ghcr.io/zarf-dev/packages/init:unset-development-only: not found
```

**Comment by ericsmalling** [comment]
For comparison, upgrading zarf via Homebrew on my mac, the 0.37.0 version works expected.
```
$ brew upgrade zarf
==> Upgrading 1 outdated package:
defenseunicorns/tap/zarf 0.36.1 -> 0.37.0
==> Fetching defenseunicorns/tap/zarf
==> Downloading https://github.com/zarf-dev/zarf/releases/download/v0.37.0/zarf_v0.37.0_Darwin_arm64
==> Downloading from https://objects.githubusercontent.com/github-production-release-asset-2e65be/398998426/efce90fa-857b-4550-875c-4c4b21e985dd?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Crede
###################################################################################################################################################################################### 100.0%
==> Upgrading defenseunicorns/tap/zarf
  0.36.1 -> 0.37.0
🍺  /opt/homebrew/Cellar/zarf/0.37.0: 4 files, 130.2MB, built in 0 seconds
==> Running `brew cleanup zarf`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
Removing: /opt/homebrew/Cellar/zarf/0.36.1... (4 files, 130.1MB)
Removing: /Users/eric/Library/Caches/Homebrew/zarf--0.36.1... (130.1MB)

$ zarf version

 NOTE  Using config file
v0.37.0
```

**Comment by ajayk** [comment]
fixed by  #25141
